modules/Microsoft.Web/sites/deploy.bicep

```
@@ -228,6 +232,47 @@ module app_authsettingsv2 'config-authsettingsv2/deploy.bicep' = if (!empty(auth
   }
 }
 
+@batchSize(1)
+module app_slots 'slots/deploy.bicep' = [for (slot, index) in slots: {
+  name: '${uniqueString(deployment().name, location)}-Slot-${slot.name}'
+  params: {
+    name: slot.name
+    appName: name
```

```suggestion
    appName: app.name
```
Reduces the dependency on the input parameter and further enables you to remove the explicit dependency.